examples/span-inspector/src/inspector.rs

```diff
@@ -0,0 +1,299 @@
+            .unwrap_or_else(|err| err.to_compile_error())
+    }
+
+    fn build_error_html(input: &str) -> String {
+        let style = String::from(
+            "
+            <style>
+                .error {
+                    whitespace = normal;
+                    color: red;
+                }
+            </style> 
+        ",
+        );
+        format!("<!DOCTYPE html><html><head>{style}</head><body><p class=\"error\">{input}</p></body></html>")
+    }
+
+    fn flatten_tokenstream(stream: TokenStream) -> Vec<TokenTree> {
+        let mut output: Vec<TokenTree> = vec![];
+        for token in stream {
+            match token {
+                TokenTree::Group(group) => {
+                    output.extend(Self::flatten_tokenstream(group.stream()));
+                }
+                other => {
+                    output.push(other);
+                }
+            }
+        }
+        output
+    }
+
+    fn build_html(input: &str, span_data: Vec<(bool, bool)>) -> String {
+        let flat_tokenstream = Self::flatten_tokenstream(TokenStream::from_str(input).unwrap());
+        let mut highlighted_string = input.to_string();
+
+        let mut token_position = span_data.len();
+        for token in flat_tokenstream.into_iter().rev() {
+            // prettyplease may insert extra tokens.
+            // This `if` statement simply ignores them.
+            let token_string = token.to_string();
+            if !matches!(token_string.as_str(), "," | "{" | "}") {
+                token_position -= 1;
+            }
+            if span_data.get(token_position).unwrap().0 {
+                highlighted_string.replace_range(
+                    token.span().byte_range(),
+                    format!(
+                        "<span class=\"highlight\">{}</span>",
+                        Self::sanitize(&token.to_string())
+                    )
+                    .as_str(),
+                );
+            } else if span_data.get(token_position).unwrap().1 {
+                highlighted_string.replace_range(
+                    token.span().byte_range(),
+                    format!(
+                        "<span class=\"generated\">{}</span>",
+                        Self::sanitize(&token.to_string())
+                    )
+                    .as_str(),
+                );
+            } else {
+                highlighted_string.replace_range(
+                    token.span().byte_range(),
+                    Self::sanitize(&token.to_string()).as_str(),
+                );
+            }
+        }
+
+        let style = String::from(
+            "
+            <style> 
+                .highlight {
+                    background-color: #ff00ff;
+                    padding: 2px 6px;
+                    border-radius: 6px;
+                }
+                .generated {
+                    color: rgba(255, 255, 255, 100);
+                    padding: 2px 6px;
+                    border-radius: 6px;
```

Comment:
These colors are hard-coded, and can be hard to read, depending on the color scheme (light/dark mode).
Ideally these would be chosen based on the current palette of the widget, but this may be hard to fix at the moment, so it's something to note for a follow-up PR.